src/utils/fill-missing-data.ts

```diff
@@ -60,3 +60,22 @@ export const fillMissingData = (data: ChartData, nrOfPointsToFill: number) => {
   }
   return data
 }
+
+const emptyArray: number[] = []
+export const transformResults = (data: ChartData, format: string) => {
+  if (format === "array" && data.format === "json") {
+    if (Array.isArray(data.result)) return data
+
+    return {
+      ...data,
+      result: (data.result.data as number[][]).reduce((acc: number[], pointData: number[]) => {
```

Comment:
```suggestion
      result: ((data as DygraphData).result.data).reduce((acc: number[], pointData: number[]) => {
```
that could be cleaner typing, or at least similar to current pattern (ChartData is an union of different *Data types), this way we know what type of data is coming from BE